File: Data Structures & Algorithms/house-robber-ii/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    find(nums, x) {
        const n = nums.length;
        let next = 0, nextnext = 0;
        let res = 0;
        const d = x === 0 ? 1 : 0
        for(let i = n - 1 - x; i >= d; i--) {
            res = Math.max(nums[i] + nextnext, next);
            nextnext = next;
            next = res;
        }
        return res;
    }
    rob(nums) {
        if(nums.length == 1) return nums[0];
        // if(nums.length === 3) {
        //     return nums[1];
        // }
        return Math.max(this.find(nums, 0), this.find(nums, 1));
    }
}
// 3 4 3
// 0 4 3 0 0
//   4 3 0 0

File: Data Structures & Algorithms/pow-x-n/submission-3.cs
public class Solution {
    public double MyPow(double x, int n) {
        double res = 1;

        if(n < 0) {
            n = -n;
            x = 1/x;
        }
        while( n > 0) {
            if(n % 2 != 0) {
                res*=x;
            }
            x *= x;
            n/=2;
        }
        return res;
    }
}
